src/Microsoft.Diagnostics.Monitoring.EventPipe/Configuration/ActivitySourceConfiguration.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Text;
+using Microsoft.Diagnostics.NETCore.Client;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe
+{
+    public sealed class ActivitySourceConfiguration : MonitoringSourceConfiguration
+    {
+        private readonly double _samplingRatio;
+        private readonly string[] _activitySourceNames;
+
+        public ActivitySourceConfiguration(
+            double samplingRatio,
+            IEnumerable<string>? activitySourceNames)
+        {
+            _samplingRatio = samplingRatio;
+            _activitySourceNames = activitySourceNames?.ToArray() ?? Array.Empty<string>();
+        }
+
+        public override IList<EventPipeProvider> GetProviders()
+        {
+            StringBuilder filterAndPayloadSpecs = new();
+            foreach (string activitySource in _activitySourceNames)
+            {
+                if (string.IsNullOrEmpty(activitySource))
+                {
+                    continue;
+                }
+
+                // Note: It isn't currently possible to get Events or Links off
+                // of Activity using this mechanism:
+                // Events=Events.*Enumerate;Links=Links.*Enumerate; See:
+                // https://github.com/dotnet/runtime/issues/102924
+
+                string sampler = string.Empty;
+
+                if (_samplingRatio < 1D)
+                {
+                    sampler = $"-ParentRatioSampler({_samplingRatio})";
+                }
+
+                filterAndPayloadSpecs.AppendLine($"[AS]{activitySource}/Stop{sampler}:-TraceId;SpanId;ParentSpanId;ActivityTraceFlags;TraceStateString;Kind;DisplayName;StartTimeTicks=StartTimeUtc.Ticks;DurationTicks=Duration.Ticks;Status;StatusDescription;Tags=TagObjects.*Enumerate;ActivitySourceVersion=Source.Version");
+            }
+
+            // Note: Microsoft-Diagnostics-DiagnosticSource only supports a
```

Comment:
We solve this problem btw with a shared string across our configurations. Since we are not planning on operating triggers and OTLP at the same time right now, it's not a concern but if we want to do all of these, we'll have to aggregate the otlp configuration with the other ones. Let's save that for the future.